### PR TITLE
Deactivate UAC action rule

### PR DESCRIPTION
**Pull request by liamtoozer**
# Motivation and Context
We need to be able to deactivate UACs on a certain date, when we no longer want to allow respondents to launch EQ.

# What has changed
* Add new queue & routing
* Rename 'wave of contact' to 'action rule' throughout
* Allow `CaseToProcessProcessor` to handle deactivate UAC action rule
* New message endpoint to handle deactivate UAC
* Remove default classifier filters

# How to test?
Run ATs.

Set up an action rule of type `DEACTIVATE_UAC`.

# Links
Trello: https://trello.com/c/MvZRwX7P

### CODE REVIEW

**Comment by liamtoozer** [comment]
[Generic Survey - UAC deactivation rule (13)](https://trello.com/c/MvZRwX7P/2323-generic-survey-uac-deactivation-rule-13)

File: src/test/resources/application-test.yml [hunk 1]
```diff
@@ -18,3 +18,4 @@ caserefgeneratorkey: rPr3qsjlU42hsDX5npfuCzlyF4UYfDTO
 queueconfig:
   consumers: 2 #this should be more than adequate to check for any multithreading problems
   retry-delay: 10  #this is for the tests around multiple receipts for the same case locking
+  rh-uac-queue: events.rh.uacUpdate
```
Comment (NickGrantONS) [inline src/test/resources/application-test.yml:21]: this needs to be deleted. Only thing that goes in this file is stuff to point at a different Rabbit/Postgres

**Review by NickGrantONS — APPROVED** [review]

File: src/test/java/uk/gov/ons/ssdc/caseprocessor/service/DeactivateUacProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  @InjectMocks private DeactivateUacProcessor underTest;
+
+  @Test
+  public void testProcessDeactivateUacRow() {
+    // Given
+    ReflectionTestUtils.setField(underTest, "outboundExchange", "testExchange");
+
+    Case caze = new Case();
+    caze.setSample(Map.of("foo", "bar"));
+    caze.setCaseRef(123L);
```
Comment (LukeLoze1) [inline src/test/java/uk/gov/ons/ssdc/caseprocessor/service/DeactivateUacProcessorTest.java:37]: Test passes without these values sample/caseRef being set.  I think I'm as guilty as anyone regarding this, but we shouldn't add to it.

**Review by LukeLoze1 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/ons/ssdc/caseprocessor/service/DeactivateUacProcessorTest.java (hunk 2 above)
Comment (NickGrantONS) [reply src/test/java/uk/gov/ons/ssdc/caseprocessor/service/DeactivateUacProcessorTest.java:37]: we'll take them out

**Review by LukeLoze1 — APPROVED** [review]

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/schedule/ActionRuleProcessor.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  private final CaseClassifier caseClassifier;
+  private final ActionRuleRepository actionRuleRepository;
+
+  public ActionRuleProcessor(
+      CaseClassifier caseClassifier, ActionRuleRepository actionRuleRepository) {
+    this.caseClassifier = caseClassifier;
+    this.actionRuleRepository = actionRuleRepository;
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW) // Start a new transaction for every WoC
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/schedule/ActionRuleProcessor.java:20]: References WoC in comment

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/schedule/ActionRuleProcessor.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  public void createScheduledActionRule(ActionRule triggeredActionRule) {
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/schedule/ActionRuleProcessor.java:21]: I suspect this is mis-named? It looks like it's triggering an existing action rule, not creating one.

File: src/test/resources/definitions.json [hunk 5]
```diff
@@ -229,6 +239,15 @@
       "arguments": {
       }
     },
+    {
+      "source": "events",
+      "vhost": "/",
+      "destination": "events.caseProcessor.deactivateUac",
+      "destination_type": "queue",
+      "routing_key": "events.caseProcessor.deactivateUac",
```
Comment (AdamHawtin) [inline src/test/resources/definitions.json:247]: I know these have already crept in other places, but I don't think the routing key shouldn't include our consuming service. It should just describe the event, e.g. `events.deactivateUac`.

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
A couple of small comments

File: src/test/resources/definitions.json (hunk 5 above)
Comment (NickGrantONS) [reply src/test/resources/definitions.json:247]: I agree, but I think that's tech debt for another ticket - needs to be sorted out consistently. People have lost/forgotten the meaning of routing keys, or conflated it with queue name

Comment (AdamHawtin) [reply src/test/resources/definitions.json:247]: Fair enough, we do need to go through putting in proper routing keys everywhere so no point enforcing it yet.

**Review by AdamHawtin — APPROVED** [review]

**Pull request merged**